### DISCUSSION TITLE
Is it possible to remove Core/Blocks supports?

### DISCUSSION THREAD

**Discussion by alexis-magina** [WordPress/gutenberg#48428] [Q&A] [answered]
Hello,
I'm searching a way to disable some features of core blocks ; 
For example, I wanted to remove the _width settings_, the _border radius_, and the _background color_ of the **core/button** block. 
Is it possible ?

I searched and tried by using theme.json, php and js filters but with no success for now.

**Comment by t-hamano** [comment, marked as answer]
Hi.

If your theme has `theme.json`, you can opt-out by setting the following. This is a sample that disables the button background color and border radius controls.

```json
{
	"$schema": "https://schemas.wp.org/trunk/theme.json",
	"version": 2,
	"settings": {
		"appearanceTools": true,
		"layout": {
			"contentSize": "840px",
			"wideSize": "1100px"
		},
		"blocks": {
			"core/button": {
				"color": {
					"background": false,
					"customGradient": false,
					"gradients": []
				},
				"border": {
					"radius": false
				}
			}
		}
	}
}
```

However, you cannot opt out of settings that are unique to that block, such as the width setting of the button block.

**Reply by ghost** [reply]
You can use [block variations](https://fullsiteediting.com/lessons/block-variations/) to disable or override block attributes, and use @t-hamano 's solution for the styling options.
